src/strands/telemetry/metrics.py

```diff
@@ -355,3 +360,53 @@ def metrics_to_string(event_loop_metrics: EventLoopMetrics, allowed_names: Optio
         A formatted string representation of the metrics.
     """
     return "\n".join(_metrics_summary_to_lines(event_loop_metrics, allowed_names or set()))
+
+
+class MetricsClient:
+    """Singleton client for managing OpenTelemetry metrics instruments.
+
+    The actual metrics export destination (console, OTLP endpoint, etc.) is configured
+    through OpenTelemetry SDK configuration by users, not by this client.
+    """
+
+    _instance = None
+    _lock = threading.Lock()
+
+    meter: Meter
+    strands_agent_invocation_count: Counter
+
+    def __init__(self) -> None:
+        """Initialize a MetricsClient instance.
+
+        Note: Initialization logic is intentionally placed in __new__ rather than __init__
+        to ensure it only runs once when the singleton instance is created, not every
+        time the class is instantiated.
+        """
+        pass
+
+    def __new__(cls) -> "MetricsClient":
```

Comment:
What's the need for a wrapper vs just doing:

```
shared_metrics = MetricsClient()
```

at the top level of the module?